lib/dialects/postgres/data-types.js

```diff
@@ -52,24 +52,24 @@ DATE.prototype.toSql = function() {
 
 var INTEGER = function() {
   if (!(this instanceof INTEGER)) return new INTEGER();
-  BaseTypes.INTEGER.apply(this, arguments);
-
-  // POSTGRES does not support length on INTEGER
-  delete this._length;
-  delete this.options.length;
```

Comment:
I'm aware, hence why i wrote the code to remove it, but it's better to always call the super method when you have inheritance code.